drivers/dac/max538x/max538x.c

```diff
@@ -0,0 +1,185 @@
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <errno.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include "max538x.h"
+#include "no_os_i2c.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+
+const struct max538x_chip_info chip_info[] = {
+	[MAX5380L] = {
+		.vfactor = 2,
+		.addr = 0x30,
+	},
+	[MAX5380M] = {
+		.vfactor = 2,
+		.addr = 0x31,
+	},
+	[MAX5380N] = {
+		.vfactor = 2,
+		.addr = 0x32,
+	},
+	[MAX5380K] = {
+		.vfactor = 2,
+		.addr = 0x33,
+	},
+	[MAX5381L] = {
+		.vfactor = 4,
+		.addr = 0x30,
+	},
+	[MAX5381M] = {
+		.vfactor = 4,
+		.addr = 0x31,
+	},
+	[MAX5381N] = {
+		.vfactor = 4,
+		.addr = 0x32,
+	},
+	[MAX5381K] = {
+		.vfactor = 4,
+		.addr = 0x33,
+	},
+	[MAX5382L] = {
+		.vfactor = 0.9,
+		.addr = 0x30,
+	},
+	[MAX5382M] = {
+		.vfactor = 0.9,
+		.addr = 0x31,
+	},
+	[MAX5382N] = {
+		.vfactor = 0.9,
+		.addr = 0x32,
+	},
+	[MAX5382K] = {
+		.vfactor = 0.9,
+		.addr = 0x33,
+	},
+};
+
+/***************************************************************************//**
+ * @brief Initializes the communication peripheral and checks if the device is
+ *        present.
+ *
+ * @param device     - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       parameters.
+ *
+ * @return status - The result of the initialization procedure.
+ *                  Example: -1 - I2C peripheral was not initialized or the
+ *                                device is not present.
+ *                            0 - I2C peripheral was initialized and the
+ *                                device is present.
+*******************************************************************************/
+int max538x_init(struct max538x_dev **device,
+		 struct max538x_init_param init_param)
+{
+	struct max538x_dev *dev;
+	int32_t status;
```

Comment:
`status` type should match the function type. Also consider using throughout the code only one naming: either `status`, either `ret`